library/tlconfig.py

```diff
@@ -55,17 +58,24 @@ def main():
     if len(module.params['value']) == 0:
         module.fail_json(msg="Value can't be empty")
 
-    if getvalue(module.params['param']) != module.params['value']:
+    current=getvalue(module.params['param'])
+    new=module.params['value']
+
+    if current != new:
         ret = setvalue(module.params['param'], module.params['value'])
 
         if ret is not 0:
             msg = "Failed setting %s: tl-config returned %d" % (module.params['param'], ret)
             module.fail_json(msg=msg)
-        changed = True
+        result['changed'] = True
+
+        if module._diff:
+            result['diff'] = dict(
```

Comment:
Why `dict()` and not the regular `{}` syntax?

Reply:
No reason, I'm not a master snake charmer :), I was reading examples which mentioned ```dict()```. I have updated the notation :)